src/Aspire.Hosting.ValKey/ValKeyBuilderExtensions.cs

```diff
@@ -0,0 +1,148 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Globalization;
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.ValKey;
+using Aspire.Hosting.Utils;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding ValKey resources to the application model.
+/// </summary>
+public static class ValKeyBuilderExtensions
+{
+    private const string ValKeyContainerDataDirectory = "/data";
+
+    /// <summary>
+    /// Adds a ValKey container to the application model.
+    /// </summary>
+    /// <example>
+    /// Use in application host
+    /// <code>
+    /// var builder = DistributedApplication.CreateBuilder(args);
+    ///
+    /// var valKey = builder.AddValKey("valKey");
+    /// var api = builder.AddProject&lt;Projects.Api&gt;("api)
+    ///                  .WithReference(valKey);
+    ///  
+    /// builder.Build().Run(); 
+    /// </code>
+    /// </example>
+    /// <example>
+    /// Use in Api with Aspire.StackExchange.Redis
+    /// <code>
+    /// var builder = WebApplication.CreateBuilder(args);
+    /// builder.AddRedisClient("valKey");
+    ///
+    /// var multiplexer = builder.Services.BuildServiceProvider()
+    ///                                   .GetRequiredService&lt;IConnectionMultiplexer&gt;();
+    /// 
+    /// var db = multiplexer.GetDatabase();
+    /// db.HashSet("key", [new HashEntry("hash", "value")]);
+    /// var value = db.HashGet("key", "hash");
+    /// </code>
+    /// </example>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
```

Comment:
Move params under summary.

Reply:
Do we need to move returns the same way?

Reply:
I tend to put the `<remarks>` and `<example>` after everything (with remarks being before examples). Its pedantic as I don't think it truly matters but I am trying to drive some consistency which has a qualify of its own :)